config/src/main/java/com/typesafe/config/impl/ConfigBeanImpl.java

```diff
@@ -163,11 +164,14 @@ private static Object getValue(Class<?> beanClass, Type parameterType, Class<?>
             return config.getList(configPropName);
         } else if (hasAtLeastOneBeanProperty(parameterClass)) {
             return createInternal(config.getConfig(configPropName), parameterClass);
+        } else if (parameterClass.isEnum()) {
+            return config.getEnum((Class<Enum>) parameterClass, configPropName);
```

Comment:
I think if you create a local variable like `Class<Enum> enumClass = (Class<Enum>) parameterClass` you can put the SuppressWarnings on the local var instead of the entire method. There may be a better way; I've been away from java for a while lately.